scipy/stats/stats.py

```diff
@@ -4153,6 +4154,160 @@ def binary_search(n, n1, n2, side):
     return oddsratio, pvalue
 
 
+def _barnard_exact_compute_gammaln_combination(n):
+    """Compute all log combination of C(n, k)"""
+    gammaln_arr = special.gammaln(np.arange(n + 1) + 1)
+    return special.gammaln(n + 1) - gammaln_arr - gammaln_arr[::-1]
+
+
+barnardExactResult = namedtuple(
+    'barnardExactResult', ('score_statistic', 'pvalue'))
```

Comment:
Please change this to [`make_dataclass`](https://docs.python.org/3/library/dataclasses.html#dataclasses.make_dataclass), adjust the examples to access the attributes by name, and this part will be all set.

Reply:
Done in 47d2d9cb4c223ad4b9825c4fe5ad90769b0cdc93. Thank you very much for you review :-)